### PR TITLE
Fix #10028: Ensure stopped RVs do not prevent overtakes

### LINKED ISSUE #10028: [Bug]: RV overtaking fails if multiple vehicles follow closely
Oldest report about this: #9236 (kept this ticket, as it is more descriptive to the problem at hand)

### Version of OpenTTD

12.2

### Expected result

When a queue of road vehicles approaches a stopped/broken-down RV, they should overtake the RV one after another.

### Actual result

Depending on where the RV to overtake is stopped/broken-down, overtaking will fail completely, and all RV in queue are stuck.

The actual test that fails the overtaking attempt is:
https://github.com/OpenTTD/OpenTTD/blob/master/src/roadveh_cmd.cpp#L796
```
	/* Are there more vehicles on the tile except the two vehicles involved in overtaking */
	return HasVehicleOnPos(od->tile, od, EnumFindVehBlockingOvertake);
```

This check does not only check and fail for vehicles in front, but also for vehicles behind on the same tile.

### Steps to reproduce

1. Load attached savegame.
    [overtake_queue.zip](https://github.com/OpenTTD/OpenTTD/files/9550609/overtake_queue.zip)
2. Unpause.
    overtake_queue1
3. Observe one side succeeding with overtaking, the other side failing.
    overtake_queue2

### PR DESCRIPTION
**Pull request by sevii1**
## Motivation / Problem

RV overtaking would fail in certain circumstances if multiple vehicles were driving closely, usually if the stopped RV was between tiles.

Closes #10028.

## Description

Added an extra condition in the check in EnumFindVehBlockingOvertake in roadveh_cmd.cpp that checks to see if the vehicle is currently attempting to overtake but was blocked by a stopped vehicle stuck between two tiles, forcing its speed to be 0, which causes the overtake to occur anyway.

## Limitations

The road vehicle may sometimes overtake in a scenario where it shouldn't, such as if another car is traveling in the opposite direction, causing a potential accident. I can't really pinpoint what exactly causes that to happen and didn't have too much time to test it, but that is one notable edge case.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by 2TallTyler** [comment]
The commit checker is failing for trailing whitespace in your comment, and because the commit message doesn't match the standard format. The PR title is correct, so just make the commit message match. 😉 

Also, have you seen #12708? It needs testing, but maybe that's something you'd be interested in.

File: src/roadveh_cmd.cpp [hunk 1]
```diff
@@ -772,7 +772,17 @@ static Vehicle *EnumFindVehBlockingOvertake(Vehicle *v, void *data)
...
+	* first ensure that the vehicle we are looking at is a road vehicle
+	* also ensure that the vehicle we are looking at is neither of the vehicles stored in OvertakeData
+	* 
+	* A block can only happen if more than two vehicles are involved, so we can skip past this if there are only two
+	*/
+	if ((v->type == VEH_ROAD && v->First() == v) && (v != od->u && v != od->v)) {
+		/* ensure that third vehicle overtaking does not get stuck in place by allowing it to overtake */
+		return (((RoadVehicle *)v)->overtaking && v->cur_speed == 0) ? v : nullptr;
+	}
+	return nullptr;
```
Comment (rubidium42) [inline src/roadveh_cmd.cpp:785]:
```suggestion
	/**
	 * First ensure that the vehicle we are looking at is a road vehicle,
	 * Also ensure that the vehicle we are looking at is neither of the vehicles stored in OvertakeData.
	 *
	 * A block can only happen if more than two vehicles are involved, so we can skip past this if there are only two.
	 */
	if ((v->type == VEH_ROAD && v->First() == v) && (v != od->u && v != od->v)) {
		/* Ensure that third vehicle overtaking does not get stuck in place by allowing it to overtake. */
		return (static_cast<RoadVehicle *>(v)->overtaking && v->cur_speed == 0) ? v : nullptr;
	}
	return nullptr;
```
